src/reactjs/components/CIRAEditor/CiraGrid.tsx

```diff
@@ -45,65 +45,65 @@ export class CiraGrid extends React.Component<CiraGridProps, CiraGridStates> {
         ciraColumnDefs.Port,
         ciraColumnDefs.Username,
         ciraColumnDefs.CommonName,
-        ciraColumnDefs.RootCertificate,
+        ciraColumnDefs.RootCertificate
       ],
       rowData: null,
-      rowSelection: "single",
+      rowSelection: 'single',
       autoGroupColumnDef: {
-        headerName: "Checkbox",
-        field: "checkbox",
-        cellRenderer: "agGroupCellRenderer",
+        headerName: 'Checkbox',
+        field: 'checkbox',
+        cellRenderer: 'agGroupCellRenderer',
         cellRendererParams: {
           checkbox: (params) => {
-            return params.node.group === false;
-          },
-        },
+            return params.node.group === false
+          }
+        }
       },
-      softSelectedDevice: undefined,
-    };
+      softSelectedDevice: undefined
+    }
   }
 
-  //listen to parent components chages to re-render the profile grid
-  componentDidUpdate(prevProps) {
-    if (this.props.updateCiraGrid != prevProps.updateCiraGrid) {
+  // listen to parent components chages to re-render the profile grid
+  componentDidUpdate (prevProps): void {
+    if (this.props.updateCiraGrid !== prevProps.updateCiraGrid) {
       this.fetchCiraConfigs().then((data) => {
         this.setState({
-          rowData: data ? data.map(config => camelCaseReshape(config, ciraDataModal)) : [],
-        });
-      });
+          rowData: isFalsy(data) ? data.map(config => camelCaseReshape(config, ciraDataModal)) : []
+        })
+      }).catch(() => console.info('error occured'))
     }
   }
 
-  onSelectionChanged = () => {
-    if (isFunc(this.props.getSelectedCiraConfigs)) {
-      this.props.getSelectedCiraConfigs(this.gridApi.getSelectedRows());
+  onSelectionChanged = (): void => {
+    if (isFalsy(isFunc(this.props.getSelectedCiraConfigs))) {
+      this.props.getSelectedCiraConfigs(this.gridApi.getSelectedRows())
     }
-  };
+  }
 
-  //Fetch the CIRA config scripts to be displayed on the grid
-  fetchCiraConfigs = async () =>
+  // Fetch the CIRA config scripts to be displayed on the grid
+  fetchCiraConfigs = async (): Promise<any> =>
     await HttpClient.get(
-      `${this.props.rpsServer}/api/v1/admin/ciraconfigs`, this.props.rpsKey
+      `${String(this.props.rpsServer)}/api/v1/admin/ciraconfigs`, this.props.rpsKey
```

Comment:
Why don't we just make rpsServer a string?

Reply:
Updated with proper null checks for prop values taken from the application URL.